File: Week_01/G20200447010278/LeetCode_283_0278.js
/*
 * @lc app=leetcode.cn id=283 lang=javascript
 *
 * [283] 移动零
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes_tmp = function (nums) {
    let zeroNums = 0;
    for (let i = 0; i < nums.length - zeroNums; i++) {
        if (nums[i] == 0) {
            nums.splice(i, 1);
            nums.push(0);
            i--;
            zeroNums++;
        }
    }

    return nums;
};

var moveZeroes = function (nums) {
    let noZeroNums = 0;
    for (let i = 0; i < nums.length; i++) {
        if (nums[i] !== 0) {
            nums[noZeroNums] = nums[i];
            if(i > noZeroNums) nums[i] = 0;
            noZeroNums++;
        }        
    }
}
// @lc code=end

File: Week_01/G20200343040278/LeetCode_283_0278.py
#
# @lc app=leetcode.cn id=283 lang=python3
#
# [283] 移动零
#

# @lc code=start
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        if nums == []:
            return 
        j = 0
        for i in range(len(nums)):
            if nums[i] != 0:
                nums[i], nums[j] = nums[j], nums[i]
                j += 1


# 初见思路1：
# 双指针法：i定位非0元素，j定位0元素
# j同时等于遍历中记录的非零元素个数，所以初始值为0即可 


# @lc code=end

